### PR TITLE
Geysers

### PR DESCRIPTION
**Pull request by ShiftyRail**
Take the previous pull request on geysers and attempt to finish it.
3 kinds of geysers: regular, thermal vent (stable) and unstable ones.

Add randomly-generated geysers scattered across the map.
Right now the mean time to generate them is 0.6 seconds, so maybe more could be generated. (I only do the immediate neighborhood of the bunkers as of now.)

Work still need to be done on the icons and eventually on geothermal energy harvesting.

### CODE REVIEW

File: maps/snaxi.dm [hunk 1]
```diff
@@ -41,9 +41,56 @@
 
 /datum/map/active/New()
 	.=..()
+	
 	research_shuttle.name = "Research and Mining Shuttle"
 	research_shuttle.req_access = list() //It's shared by miners and researchers, so remove access requirements
 
+// Making nodes every 5*5 tiles in a 100*100 radius
+// This makes (100*100)/(10*10) = 100 nodes
```
Comment (utilitas-) [inline maps/snaxi.dm:49]: It'd be better to make this a define, either as a variable that gets unpacked or as the equation in lambda notiation at the beginning of the file, so it isn't some magic number pulled outta nowhere. Much easier to debug and adjust. 100 could be too many, too little- or the number itself could be randomized.
And is snaxi.dm _still_ called "Datum Boxstation" up top? Jesus christ, how lazy can you get?

Comment (ShiftyRail) [reply maps/snaxi.dm:49]: Well that last part wasn't me

Comment (utilitas-) [reply maps/snaxi.dm:49]: I know. Someone should make a pull request to change that _just_ to rub it in.

File: maps/_map_override.dm [hunk 2]
```diff
@@ -46,5 +46,9 @@
 		#undef MAP_OVERRIDE
 		#include "wheelstation.dm"
 		#define MAP_OVERRIDE 8
+	#elif MAP_OVERRIDE == 9
+		#undef MAP_OVERRIDE
+		#include "snaxi.dm"
+		#define MAP_OVERRIDE 9
```
Comment (N3X15) [inline maps/_map_override.dm]: uhhh

File: maps/snaxi.dm [hunk 3]
```diff
@@ -41,9 +41,56 @@
...
+			CHECK_TICK
+	for (var/x = center_x; x >= center_x - center_x/3; x = x - 10)
+		for (var/y = center_y; y <= center_y + center_y/3; y = y + 10)
+			gaussian_geyer(x, y)
+			CHECK_TICK
+		for (var/y = center_y; y >= center_y - center_y/3; y = y - 10)
+			gaussian_geyer(x, y)
+			CHECK_TICK
+
+/proc/gaussian_geyer(var/x, var/y)
```
Comment (N3X15) [inline maps/snaxi.dm:66]: `geyer`

**Review by N3X15 — APPROVED** [review]
Nothing else jumps out at me

**Comment by utilitas-** [comment]
Wait, why is the sprite in beach.dmi? Can it be applied to sand as well? If not, just put it in a seperate file. Make a snow.dmi.

**Comment by ShiftyRail** [comment]
The sprite really isn't that good (not the good colour).
I could codersprite it if that's really needed.

File: code/ATMOSPHERICS/hvac/geyser.dm [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	var/preparing_smoke = 0
+
+/obj/structure/geyser/process()
+	if (preparing_smoke)
+		return // We're about to exhaust smoke
+	if(prob(smoke_probability))
+		puff_smoke()
+	var/turf/T = loc
+	if(!istype(T))
+		return //Not on a turf
```
Comment (Zth--) [inline code/ATMOSPHERICS/hvac/geyser.dm:28]: Is there any reason you check for a null turf _after_ the _puff_smoke()_ call?

Comment (ShiftyRail) [reply code/ATMOSPHERICS/hvac/geyser.dm:28]: Gerysers are `obj` and can, probably, be put inside other atoms in specific situations.

Comment (Zth--) [reply code/ATMOSPHERICS/hvac/geyser.dm:28]: But would that make any sense? I just don't see it

**Comment by coldcola** [comment]
this is broken

**Pull request merged**